orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -100,7 +100,14 @@ class DependentPipelineStarter implements ApplicationContextAware {
 
     def trigger = pipelineConfig.trigger
     //keep the trigger as the preprocessor removes it.
-    def expectedArtifacts = pipelineConfig.expectedArtifacts
+
+    for (ExecutionPreprocessor preprocessor : executionPreprocessors.findAll {
+      it.supports(pipelineConfig, ExecutionPreprocessor.Type.PIPELINE)
+    }) {
+      pipelineConfig = preprocessor.process(pipelineConfig)
+    }
+
+    pipelineConfig.trigger = trigger
```

Comment:
Any reason you are moving this back? It breaks what I fixed in #2963.

Reply:
Apparently it will be required to run resolve artifacts twice. I added a comment in code explaining why